src/nxt_http_request.c

```diff
@@ -866,6 +862,63 @@ nxt_http_request_close_handler(nxt_task_t *task, void *obj, void *data)
 }
 
 
+static nxt_int_t
+nxt_http_request_access_log(nxt_task_t *task, nxt_http_request_t *r)
+{
+    nxt_int_t          ret;
+    nxt_str_t          str;
+    nxt_bool_t         enable;
+    nxt_router_conf_t  *rtcf;
+
+    rtcf = r->conf->socket_conf->router_conf;
+
+    r->logged = 1;
+
+    if (rtcf->access_log == NULL) {
+        return NXT_OK;
+    }
+
+    enable = 1;
+
+    if (rtcf->log_enable != NULL) {
+
+        if (nxt_tstr_is_const(rtcf->log_enable)) {
+            nxt_tstr_str(rtcf->log_enable, &str);
+
+        } else {
+            rtcf = r->conf->socket_conf->router_conf;
+
+            ret = nxt_tstr_query_init(&r->tstr_query, rtcf->tstr_state,
+                                      &r->tstr_cache, r, r->mem_pool);
+            if (nxt_slow_path(ret != NXT_OK)) {
+                return NXT_ERROR;
+            }
+
+            nxt_tstr_query(task, r->tstr_query, rtcf->log_enable, &str);
+
+            if (nxt_slow_path(nxt_tstr_query_failed(r->tstr_query))) {
+                return NXT_ERROR;
+            }
+        }
+
+        if (str.length == 0
+            || nxt_str_eq(&str, "0", 1)
+            || nxt_str_eq(&str, "false", 5)
+            || nxt_str_eq(&str, "null", 4)
+            || nxt_str_eq(&str, "undefined", 9))
+        {
+            enable = 0;
+        }
+    }
+
+    if (enable ^ rtcf->log_negative) {
```

Comment:
Struggling with the logic here... truth table time!

```
enable ^ log_negative
0 ^ 0 | 0
0 ^ 1 | 1
1 ^ 0 | 1
1 ^ 1 | 0
```

At the moment, I can't say that makes any sense to me... (in terms of when and when not to log)

Ah! It represents the '!' to negate the log condition. OK, can we rename it to _log_negation_? Then the second entry in the above table makes sense...

Reply:
I'm fine with `log_negative`, `log_negation`, and `log_negate`. Welcome to check again.

Here's another use case in the http route matching, we used the math operation `^`.
https://github.com/nginx/unit/blob/master/src/nxt_http_route.c#L1874

I agree it's more readable to write like `negate ? !expr : expr`, but XOR `^` may offter better performance than conditional logic `?` due to its non-branching nature.

Reply:
log_negate is fine.

No problem with the use of **^**